src/overlays/actors/ovl_Obj_Switch/z_obj_switch.c

```diff
@@ -5,17 +5,79 @@
  */
 
 #include "z_obj_switch.h"
+#include "objects/gameplay_dangeon_keep/gameplay_dangeon_keep.h"
 
 #define FLAGS 0x00000010
 
 #define THIS ((ObjSwitch*)thisx)
 
+#define COS_OF_5_PI_DIV_8 -0.38268343f
+
+#define OBJ_SWITCH_GET_TYPE(p) ((p)&7)
+#define OBJ_SWITCH_GET_SUBTYPE(p) (p >> 4 & 7)
+#define OBJ_SWITCH_GET_SWITCH_FLAG(p) ((p) >> 8 & 0x7F)
+#define OBJ_SWITCH_IS_FROZEN(p) ((p) >> 7 & 1)
+#define OBJ_SWITCH_UNSET_FROZEN(p) ((p) &= ~(1 << 7))
+#define OBJ_SWITCH_IS_INVISIBLE(p) ((p) >> 3 & 1)
+#define OBJ_SWITCH_GET_COLOR_ID(z) ((z)&1)
```

Comment:
Current standard is params accesses to take in the base actor pointer, as well as be paced in `*.h` file.